pyomo/contrib/appsi/base.py

```diff
@@ -1406,17 +1406,17 @@ def update(self, timer: HierarchicalTimer = None):
             vars_to_update = list()
             for v in vars_to_check:
                 _v, lb, ub, fixed, domain_interval, value = self._vars[id(v)]
-                if lb is not v._lb:
-                    vars_to_update.append(v)
-                elif ub is not v._ub:
-                    vars_to_update.append(v)
-                elif (fixed is not v.fixed) or (fixed and (value != v.value)):
+                if (fixed is not v.fixed) or (fixed and (value != v.value)):
                     vars_to_update.append(v)
                     if self.update_config.treat_fixed_vars_as_params:
                         for c in self._referenced_variables[id(v)][0]:
                             cons_to_remove_and_add[c] = None
                         if self._referenced_variables[id(v)][2] is not None:
                             need_to_set_objective = True
+                elif lb is not v._lb:
+                    vars_to_update.append(v)
+                elif ub is not v._ub:
+                    vars_to_update.append(v)
```

Comment:
This catches if the user re-assigns the lower / upper bound, but what about bounds that include mutable Params?  Does changing the Param value get propagated to updated bounds?

Reply:
that should get caught on the next two lines.

Reply:
I don't think that's the case: that only catches if the *domain* changes, and not if the value of the bounds change:
```python
>>> from pyomo.environ import *
>>> m = ConcreteModel()
>>> m.x = Var()
>>> m.p = Param(mutable=True, initialize=1)
>>> m.x.bounds = -m.p, m.p
>>> m.x.bounds
(-1.0, 1.0)
>>> m.x.domain.get_interval()
(None, None, 0)
>>> m.p = 5
>>> m.x.bounds
(-5.0, 5.0)
>>> m.x.domain.get_interval()
(None, None, 0)
```

Reply:
oh, my mistake. You are right. The case of bounds with mutable parameters is actually caught with the `update_params` method and is handled differently for each solver. [Here](https://github.com/Pyomo/pyomo/blob/c471729e7b9fd6236e88e1913760f1fa4f2be121/pyomo/contrib/appsi/solvers/gurobi.py#L430) is how it is handled with Gurobi.